lib/elixir/lib/stream.ex

```diff
@@ -1137,11 +1137,49 @@ defmodule Stream do
     lazy(enum, %{}, fn f1 -> R.uniq_by(fun, f1) end)
   end
 
+  @doc """
+  Builds a stream from an index, either starting from offset, or given by function.
+
+  May receive a function or an integer offset.
+
+  If an `offset` is given, it will emit elements from offset.
+
+  If a `function` is given, it will invoke the function with
+  elements from offset.
```

Comment:
Maybe this reads better, just throwing it out there (I like lists in technical docs, easier to skim):

```suggestion
  `fun_or_offset` can be either an integer, or a function:

    * If `fun_or_offset` is an integer, the returned stream will emit consecutive
      integers starting from the given offset.

    * If `fun_or_offset` is a function, the stream will call the function on
      consecutive integers starting from `0`. The returned stream will emit the
      elements returned by the function.
```